sdk/storage/inc/datalake/datalake_utilities.hpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "common/crypt.hpp"
+
+#include <map>
+#include <string>
+
+namespace Azure { namespace Storage { namespace DataLake {
+
+  inline std::map<std::string, std::string> DeSerializeMetadata(
+      const std::string& dataLakePropertiesString)
+  {
+    std::map<std::string, std::string> result;
+
+    std::string::const_iterator cur = dataLakePropertiesString.begin();
+
+    while (cur != dataLakePropertiesString.end())
+    {
+      auto key_begin = cur;
+      auto key_end = std::find(cur, dataLakePropertiesString.end(), '=');
+      std::string key = std::string(key_begin, key_end);
+      cur = key_end;
+      if (cur != dataLakePropertiesString.end())
+      {
+        ++cur;
+      }
+
+      auto value_begin = cur;
+      auto value_end = std::find(cur, dataLakePropertiesString.end(), ',');
+      std::string value = std::string(value_begin, value_end);
+      cur = value_end;
+      if (cur != dataLakePropertiesString.end())
+      {
+        ++cur;
+      }
+
+      if (!key.empty() || !value.empty())
+      {
+        result[std::move(key)] = Base64Decode(value);
+      }
+    }
+
+    return result;
+  }
+
+  inline std::string SerializeMetadata(
+      const std::map<std::string, std::string>& dataLakePropertiesMap)
+  {
+    std::string result;
+    for (const auto& pair : dataLakePropertiesMap)
+    {
+      result.append(pair.first + "=" + Base64Encode(pair.second) + ",");
+    }
+    result.pop_back();
```

Comment:
What if the input map is empty? This line will crash the program.

Reply:
Added an `if`.